collections-lists/ConvertIAsyncEnumerableToList/ConvertIAsyncEnumerableToListLibrary/UserPageFetcher.cs

```diff
@@ -0,0 +1,49 @@
+﻿using ConvertIAsyncEnumerableToListLibrary.Model;
+using System.Runtime.CompilerServices;
+
+namespace ConvertIAsyncEnumerableToListLibrary;
+public static class UserPageFetcher
+{
+    public static async IAsyncEnumerable<User> GetUsersAsync()
+    {
+        for (int page = 1; page <= 3; page++)
+        {
+            await Task.Delay(500);
+
+            var users = new List<User>
+            {
+                new User { Id = 1, Name = "Alice" },
+                new User { Id = 2, Name = "Bob" },
+                new User { Id = 3, Name = "John" },
```

Comment:
Maybe we should adjust these Id values to be related to the page value from the for loop, so that we don't have the same Id each time, even if we have repeated names.